src/hooks/users.ts

```diff
@@ -1,71 +1,106 @@
-import { useMemo } from "react";
+import { useCallback, useEffect, useMemo } from "react";
+import { chunk } from "lodash";
 
 import { universeUsersSelector, usersByIdSelector } from "utils/selectors";
-import { WithId } from "utils/id";
+import { WithId, withId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
 import {
   useSparkleFirestoreConnect,
   isLoaded,
+  SparkleRFQConfig,
 } from "hooks/useSparkleFirestoreConnect";
-import { useConnectCurrentVenueNG } from "hooks/useConnectCurrentVenueNG";
 import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
 
 import { User } from "types/User";
+import {
+  useFirestore,
+  useFirestoreConnect,
+  WhereOptions,
+} from "react-redux-firebase";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
 
-// NOTE: Heuristic approach. Way faster and efficient.
-// Is based on assumption that there is only one parent venue and it is an entry point for any user
+// NOTE: Logically correct version of pulling users
 export const useConnectUniverseUsers = () => {
   const venueId = useVenueId();
+  const firestore = useFirestore();
 
-  const { currentVenue, isCurrentVenueLoaded } = useConnectCurrentVenueNG(
-    venueId
-  );
-
-  useSparkleFirestoreConnect(() => {
-    if (!isCurrentVenueLoaded || !currentVenue || !venueId) return [];
-
-    const queryVenueId = currentVenue.parentId || venueId;
-    return [
-      {
-        collection: "users",
-        where: ["enteredVenueIds", "array-contains", queryVenueId],
-        storeAs: "universeUsers",
-      },
-    ];
-  });
-};
-
-// NOTE: Logically correct version of pulling users
-export const useConnectUniverseUsers_V2 = () => {
-  const venueId = useVenueId();
+  const { currentVenue, isCurrentVenueLoaded } = useConnectCurrentVenueNG();
 
   const { relatedVenues, isRelatedVenuesLoaded } = useConnectRelatedVenues({
     venueId,
   });
 
-  useSparkleFirestoreConnect(() => {
+  // useEffect(() => {
+  //   if (!isRelatedVenuesLoaded) return;
+
+  //   const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
+
+  //   const chunkedListeners = chunk(universeUserVenueIds, 10).map<
+  //     SparkleRFQConfig
+  //   >((venueIdsChunk) => ({
+  //     collection: "users",
+  //     where: ["enteredVenueIds", "array-contains", venueIdsChunk[0]],
```

Comment:
Guessing you added `venueIdsChunk[0]` to get just 1 `venueId` as a hack to play with using `array-contains` instead of `array-contains-any`. But just wanted to explicitly call out that this code as-is will ignore 9 of the `venueId`s in each 'chunk'.